src/app/forms/custom-sections/tyres/tyres.component.ts

```diff
@@ -21,34 +36,115 @@ export class TyresComponent implements OnInit, OnDestroy, OnChanges {
   public form!: CustomFormGroup;
   private _formSubscription = new Subscription();
 
-  constructor(public dfs: DynamicFormService) {}
+  constructor(public dfs: DynamicFormService, private referenceDataService: ReferenceDataService) {}
 
   ngOnInit(): void {
-    const template = getTyresSection(this.vehicleTechRecord.vehicleType);
+    this.form = this.dfs.createForm(this.template, this.vehicleTechRecord) as CustomFormGroup;
+    this._formSubscription = this.form.cleanValueChanges.pipe(debounceTime(400)).subscribe(event => {
+      this.formChange.emit(event);
+    });
 
-    this.form = this.dfs.createForm(template, this.vehicleTechRecord) as CustomFormGroup;
-
-    this._formSubscription = this.form.cleanValueChanges.pipe(debounceTime(400)).subscribe(event => this.formChange.emit(event));
+    this.referenceDataService.loadReferenceData(ReferenceDataResourceType.Tyres);
   }
 
-  ngOnChanges() {
+  ngOnChanges(simpleChanges: SimpleChanges) {
+    const { vehicleTechRecord } = simpleChanges;
+
+    if (vehicleTechRecord.firstChange === false) {
+      const currentAxles = vehicleTechRecord.currentValue.axles;
+      const previousAxles = vehicleTechRecord.previousValue.axles;
+
+      currentAxles.forEach((value: Axle, index: number) => {
+        if (value.tyres?.fitmentCode !== previousAxles[index].tyres.fitmentCode) {
+          this.getTyresRefData(value.tyres!, value.axleNumber!);
+        }
+      });
+    }
+
     this.form?.patchValue(this.vehicleTechRecord, { emitEvent: false });
   }
 
   ngOnDestroy(): void {
     this._formSubscription.unsubscribe();
   }
 
+  get template() {
+    switch (this.vehicleTechRecord.vehicleType) {
+      case VehicleTypes.PSV:
+        return tyresTemplatePsv;
+      case VehicleTypes.HGV:
+        return tyresTemplateHgv;
+      case VehicleTypes.TRL:
+        return tyresTemplateTrl;
+    }
+  }
+
+  get isPsv(): boolean {
+    return this.vehicleTechRecord.vehicleType === VehicleTypes.PSV;
+  }
+
+  get isHgvOrTrl(): boolean {
+    return this.vehicleTechRecord.vehicleType === VehicleTypes.HGV || this.vehicleTechRecord.vehicleType === VehicleTypes.TRL;
+  }
+
   get types(): typeof FormNodeEditTypes {
     return FormNodeEditTypes;
   }
 
+  get widths(): typeof FormNodeWidth {
+    return FormNodeWidth;
+  }
+
+  get speedCategorySymbol(): MultiOptions {
+    return speedCategorySymbolAsOptions();
+  }
+
+  get fitmentCode(): MultiOptions {
+    return fitmentCodeAsOptions();
+  }
+
   get axles(): CustomFormArray {
     return this.form.get(['axles']) as CustomFormArray;
   }
 
+  getAxleTyres(i: number): CustomFormGroup {
+    return this.axles.get([i, 'tyres']) as CustomFormGroup;
+  }
+
+  getTyresRefData(tyre: Tyres, axleNumber: number) {
```

Comment:
Could we get some tests for these new methods please, I know they're likely to change in terms of the source of data but would be good to assure it works.

Reply:
I knew I had forgot to do something lol, yeah sure!